### PR TITLE
(PoC) Kernel: Implement unveil() as a prefix-tree

### PR DESCRIPTION
**Pull request by alimpfard**
For #4530.

### CODE REVIEW

**Comment by bgianfo** [comment]
Does it make sense to add new unit tests targetting the new AK/Trie.h implementation directly?

**Comment by alimpfard** [comment]
Ah definitely. thanks for reminding me~

**Comment by awesomekling** [comment]
This looks very nice!

**Comment by alimpfard** [comment]
I don't think there's anything wrong with it functionality-wise, basic-ish testing with the (included) `test-unveil` does show reasonable behaviour, so I don't think there's anything else blocking it from my side.

File: AK/Tests/TestTrie.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    Array<int, 64> input;
+    for (size_t i = 0; i < input.size(); ++i)
+        input[i] = i;
+
+    // Iteration order is preorder (order between adjacent nodes is not defined, but parents come before children)
+    // in this case, the tree is linear.
+    size_t i = 0;
+    for (auto& node : bunch_of_numbers) {
+        EXPECT_EQ(input[i], node.value());
+        ++i;
```
Comment (alimpfard) [inline AK/Tests/TestTrie.cpp:79]: :facepalm:
This doesn't test anything damnit

**Pull request merged**